astropy/coordinates/sky_coordinate.py

```diff
@@ -41,6 +42,24 @@ def FRAME_ATTR_NAMES_SET():
     return out
 
 
+class SkyCoordColumnInfo(ColumnInfo):
+    def __init__(self, *args, **kwargs):
+        super(SkyCoordColumnInfo, self).__init__(*args, **kwargs)
+        self._format_cache = {}
+        if self._parent_col is not None:
+            data = self._parent_col().data
+            self.unit = ','.join(str(getattr(data, comp).unit) for comp in data.components)
+
+    def default_format_func(self, format, val):
```

Comment:
Here, clearly a function is useful, but I'm a bit confused about the need for a cache. Is this for the case that one changes the representation half-way through, and then perhaps change it back? Since this quite a list comprehension loop, is this worth it? Could one just calculate it on the fly?

Separately, I think it would in principle be nice if `col.info.default_format` gave a string that would give a good clue on how to set `format` to do something else. Could one do this using my `default_format` suggestion and implement it as a property as follows?
```
@property
def default_format(self):
    if self.parent_col is None:
       raise ValueError("Default format is undefined for a SkyCoordColumnInfo "
                        "instance not attached to a SkyCoordObject.")
    formats = ['{0.' + compname + '.value:}' for compname
               in self._parent_col().data.components]
    return ' '.join(formats)
```